examples/plot_connectivity.py

```diff
@@ -62,81 +63,72 @@
 ###############################################################################
 # We can also define our own connections to test the effect of different
 # connectivity patterns. To start, ``net.clear_connectivity()`` can be used
-# to clear all cell to cell connections. By default, previously defined drives
+# to clear all cell-to-cell connections. By default, previously defined drives
 # to the network are retained, but can be removed with ``net.clear_drives()``.
 # ``net.add_connection`` is then used to create a custom network. Let us first
 # create an all-to-all connectivity pattern between the L5 pyramidal cells,
 # and L2 basket cells. :meth:`hnn_core.Network.add_connection` allows
-# connections to be specified with either cell names, or the gids directly.
-# If multiple gids are provided for either the sources or the targets,
-# they will be connected in an all-to-all pattern.
-
-net_all = default_network(params, add_drives_from_params=True)
-net_all.clear_connectivity()
-
-# Pyramidal cell connections
-location, receptor = 'distal', 'ampa'
-weight, delay, lamtha = 1.0, 1.0, 70
-src = 'L5_pyramidal'
-for target in ['L5_pyramidal', 'L2_basket']:
-    net_all.add_connection(src, target, location, receptor,
-                           delay, weight, lamtha)
-
-# Basket cell connections
-location, receptor = 'soma', 'gabaa'
-weight, delay, lamtha = 1.0, 1.0, 70
-src = 'L2_basket'
-for target in ['L5_pyramidal', 'L2_basket']:
-    net_all.add_connection(src, target, location, receptor,
-                           delay, weight, lamtha)
+# connections to be specified with either cell names, or the cell IDs (gids)
+# directly.
+
+
+def get_network(probability=1.0):
+    net = default_network(params, add_drives_from_params=True)
+    net.clear_connectivity()
+
+    # Pyramidal cell connections
+    location, receptor = 'distal', 'ampa'
+    weight, delay, lamtha = 1.0, 1.0, 70
+    src = 'L5_pyramidal'
+    for target in ['L5_pyramidal', 'L2_basket']:
+        net.add_connection(src, target, location, receptor,
+                           delay, weight, lamtha, probability=probability)
+
+    # Basket cell connections
+    location, receptor = 'soma', 'gabaa'
+    weight, delay, lamtha = 1.0, 1.0, 70
+    src = 'L2_basket'
+    for target in ['L5_pyramidal', 'L2_basket']:
+        net.add_connection(src, target, location, receptor,
+                           delay, weight, lamtha, probability=probability)
+    return net
+
 
+net_all = get_network()
```

Comment:
another problem with this example is that it's too slow. At some point, we'll start hitting the 2 hour limit of CircleCI. I am trying to anticipate and delay that as much as possible. Any way we can make it run fewer simulations without diluting the message @ntolley ? Perhaps leave out the first simulation?

Reply:
Agreed, the first simulation isn't really necessary. This PR isn't teaching much besides how to change the connectivity, the comparisons just demonstrate "hey that's different". Once I've spent more time exploring these connectivity patterns I can update to a more meaningful example.